src/vrep_plugin.cpp

```diff
@@ -782,6 +786,72 @@ static mc_control::MCGlobalController & gc()
   return *gc_ptr;
 }
 
+static void setup_datastore()
+{
+  auto & ctl = gc_ptr->controller();
+  auto add_datastore_entries = [](mc_control::MCController & ctl) {
```

Comment:
Why the lambda?

Reply:
Ideally we should add the datastore entries to all enabled controllers. But then I realized we only have access to the current controller from outside MCGlobalController, so for now I just did it for the first controller.

Reply:
Yes. This is a defect in `MCGlobalController` current design. Ideally it should be notified when the controller changes so it can do this kind of setup.